docs/explanations/architecture/styles.md

```diff
@@ -0,0 +1,453 @@
+
+### 3. Global styles
+
+Global Styles refers to a mechanism that generates site-wide styles. Unlike the block styles described in the previous section, these are not serialized into the post content, they're not attached to the block HTML. Instead, the output of this system is a new stylesheet with id `global-styles-inline-css`.
+
+This mechanism was [introduced in WordPress 5.8](https://make.wordpress.org/core/2021/06/25/introducing-theme-json-in-wordpress-5-8/). At the time, it only took data from WordPress and the active theme. WordPress 5.9 expanded the system to also take style data from users.
+
+This is the general data flow:
+
+![global-styles-input-output](./assets/global-styles-input-output.png)
+
+The process of generating the stylesheet has, in essence, three steps:
+
+1. Gather data: the `theme.json` file [bundled with WordPress](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/theme.json)), the `theme.json` file of the active theme if it exists, and the user's styles provided via the global styles UI in the site editor.
+2. Consolidate data: the structured information coming from different origins -WordPress defaults, theme, and user- is normalized and merged into a single structure.
+3. Convert data into a stylesheet: convert the internal representation into CSS style rules and enqueue them as a stylesheet.
+
+#### Gather data
+
+The data can come from three different origins: WordPress defaults, the active theme, or the user. All three of them use the same [`theme.json` format](https://developer.wordpress.org/block-editor/reference-guides/theme-json-reference/).
+
+ Data from WordPress and the active theme is retrieved from the corresponding `theme.json` file. Data from the user is pulled from the database, where it's stored after the user saves the changes they did via the global styles sidebar in the site editor.
+
+#### Consolidate data
+
+The goal of this phase is to build a consolidated structure.
+
+There's two important processes going on in this phase. First, the system needs to normalize all the incoming data, as different origins may be using different versions of the `theme.json` format. For example, a theme may be using [v1](https://developer.wordpress.org/block-editor/reference-guides/theme-json-reference/theme-json-v1/) while the WordPress base is using [the latest version](https://developer.wordpress.org/block-editor/reference-guides/theme-json-reference/theme-json-living/). Second, the system needs to decide how to merge the input into a single structure. This will be the focus of the the following sections.
```

Comment:
```suggestion
There are two important processes going on in this phase. First, the system needs to normalize all the incoming data, as different origins may be using different versions of the `theme.json` format. For example, a theme may be using [v1](https://developer.wordpress.org/block-editor/reference-guides/theme-json-reference/theme-json-v1/) while the WordPress base is using [the latest version](https://developer.wordpress.org/block-editor/reference-guides/theme-json-reference/theme-json-living/). Second, the system needs to decide how to merge the input into a single structure. This will be the focus of the following sections.
```